### ISSUE TITLE
CI failing since tools update

### ISSUE THREAD

**Issue by Thomas1664** [microsoft/vcpkg#23284]
**Describe the bug**
All CIs except OSX and Linux are failing on stage "Test modified ports..." if PR branch is not even with master. If they are even, only `arm64_windows` and `x64_windows_static_md` fail. These bugs are probably introduced by PRs #23162 and #22985. At least 4 PRs experienced the exact same CI failure after those commits. 

**Environment**
CI

**To Reproduce**
Trigger CI run

**Failure logs** 
```
A suitable version of 7zip was not found (required v19.0.0). Downloading portable 7zip v19.0.0...
Extracting 7zip...
msiexec failed while extracting 'D:\downloads\7z1900-x64.msi' with message:
T
##[error]vcpkg ci failed
At C:\a\1\s\scripts\azure-pipelines\test-modified-ports.ps1:178 char:5
+     throw "vcpkg ci failed"
+     ~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : OperationStopped: (vcpkg ci failed:String) [], RuntimeException
+ FullyQualifiedErrorId : vcpkg ci failed
##[error]PowerShell exited with code '1'.
```

CC: @BillyONeal @JackBoosY

**Comment by Thomas1664** [comment]
Update: if a PR is in sync with master only arm64_windows and x64_windows_static_md fail

**Comment by dg0yt** [comment]
I interprete the logging output as indicating an implicit fetch of 7zip when fetching one of the other tools. I added an explicit early fetch of 7zip in #23284, complementing the explicit fetches of cmake, ninja, git. This seems to fix the issue in CI.

IIRC we saw similar logging output in a previous issue with installing 7zip.

**Comment by dg0yt** [comment]
> IIRC we saw similar logging output in a previous issue with installing 7zip.

Here: #19490

**Comment by Neumann-A** [comment]
The error is:
```
[DEBUG] 7532: cmd_execute_and_stream_data() returned 1601 after    48467 us
The Windows Installer service could not be accessed. Contact your support personnel to verify that the Windows Installer service is properly registered.
```
so it is unrelated to the mentioned PRs but a problem with the runners.

**Comment by BillyONeal** [comment]
Yeah I think we got something pushed to our VMs that breaks msiexec :(

**Comment by BillyONeal** [comment]
https://github.com/microsoft/azure-pipelines-agent/releases/tag/v2.200.2 I observe this release was pushed around the same time

**Comment by BillyONeal** [comment]
https://github.com/microsoft/azure-pipelines-agent/issues/3770

**Comment by Thomas1664** [comment]
Now installation of Python 2 fails with the following error:
```
-- Downloading https://www.python.org/ftp/python/2.7.18/python-2.7.18.amd64.msi -> python-2.7.18.amd64.msi...
T h e   W i n d o w s   I n s t a l l e r   S e r v i c e   c o u l d   n o t   b e   a c c e s s e d .   T h i s   c a n   o c c u r   i f   t h e   W i n d o w s   I n s t a l l e r   i s   n o t   c o r r e c t l y   i n s t a l l e d .   C o n t a c t   y o u r   s u p p o r t   p e r s o n n e l   f o r   a s s i s t a n c e . 
 
 CMake Error at scripts/cmake/vcpkg_find_acquire_program.cmake:692 (message):
  Unable to find PYTHON2
```

This seems to be also caused by msi installer because of the UTF16 encoding (spaces between letters)

Observed on CI by installing cppcms and qtwebengine for triplet arm64-windows

CC @BillyONeal

**Comment by Neumann-A** [comment]
Just the same failure as before. Nothing new to see here. Previously it failed to extract 7z.mis which was replaced by tar on windows and now the same failure happens with python2.msi.  This will be solved by #23431

**Comment by Thomas1664** [comment]
This is fixed now
